src/FNA3D_Driver_Vulkan.c

```diff
@@ -8683,6 +8995,9 @@ static void VULKAN_ApplyEffect(
 	renderer->currentEffect = effectData;
 	renderer->currentTechnique = technique;
 	renderer->currentPass = pass;
+
+	renderer->vertexSamplerDescriptorSetDataNeedsUpdate = 1;
```

Comment:
This will need to be at the top of the function, since there are returns prior to the end of the function that can change shader state.